tls/s2n_ktls.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+
+#include "utils/s2n_safety.h"
+
+struct s2n_connection;
+
+/* A set of kTLS configurations representing the combination of sending
+ * and receiving.
+ */
+typedef enum {
+    /* Disable kTLS. */
+    S2N_KTLS_MODE_DISABLED = 0,
+    /* Enable kTLS for the send socket. */
+    S2N_KTLS_MODE_SEND = 1,
+    /* Enable kTLS for the recv socket. */
+    S2N_KTLS_MODE_RECV = 2,
+    /* Enable kTLS for both rx and tx socket. */
+    S2N_KTLS_MODE_DUPLEX = 3,
+} s2n_ktls_mode;
+
+S2N_RESULT s2n_ktls_is_mode(s2n_ktls_mode original_state, s2n_ktls_mode check_state, bool *is_true);
+
+int s2n_config_ktls_enable(struct s2n_config *config, s2n_ktls_mode ktls_mode);
+S2N_RESULT s2n_config_is_ktls_requested(struct s2n_config *config, s2n_ktls_mode ktls_mode, bool *requested);
+S2N_RESULT s2n_connection_set_ktls_enabled(struct s2n_connection *conn, s2n_ktls_mode mode);
+S2N_RESULT s2n_connection_matches_ktls_mode(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, bool *matches);
```

Comment:
Are these APIs supposed to be private/return S2N_RESULT? I thought `s2n_connection_matches_ktls_mode` needed to be public?

Reply:
I am going to remove these functions until I actually make use of them in the future